crates/webidl/src/main.rs

```diff
@@ -0,0 +1,114 @@
+use anyhow::{Context, Result};
+use sourcefile::SourceFile;
+use std::ffi::OsStr;
+use std::fs;
+use std::path;
+use std::process::Command;
+
+fn unwrap_not_found(err: std::io::Result<()>) -> std::io::Result<()> {
+    match err {
+        Ok(()) => Ok(()),
+        Err(err) => match err.kind() {
+            std::io::ErrorKind::NotFound => Ok(()),
+            _ => Err(err),
+        },
+    }
+}
+
+/// Read all WebIDL files in a directory into a single `SourceFile`
+fn read_source_from_path(dir: &path::Path) -> Result<SourceFile> {
+    let entries = fs::read_dir(dir).context("reading webidls directory")?;
+    let mut source = SourceFile::default();
+    for entry in entries {
+        let entry = entry.context(format!("getting {}/*.webidl entry", dir.display()))?;
+        let path = entry.path();
+        if path.extension() != Some(OsStr::new("webidl")) {
+            continue;
+        }
+        source = source
+            .add_file(&path)
+            .with_context(|| format!("reading contents of file \"{}\"", path.display()))?;
+    }
+
+    Ok(source)
+}
+
+fn main() -> Result<()> {
+    #[cfg(feature = "env_logger")]
+    env_logger::init();
+
+    match std::env::args().into_iter().collect::<Vec<_>>().as_slice() {
+        [ _, from, to ] => {
```

Comment:
This seems like a bit of a wonky way to "parse" command line arguments, could this just make an iterator and use `next().unwrap()`? We don't really need super-nice error messages here and I think it's worth saving two levels of indentation if we can.